pkg/controller/deployment/deployment_controller_test.go

```diff
@@ -267,6 +282,30 @@ func TestSyncDeploymentDontDoAnythingDuringDeletion(t *testing.T) {
 	f.run(getKey(d, t))
 }
 
+func TestSyncDeploymentDeletionRace(t *testing.T) {
+	f := newFixture(t)
+
+	d := newDeployment("foo", 1, nil, nil, nil, map[string]string{"foo": "bar"})
+	d2 := *d
+	// Lister (cache) says NOT deleted.
+	f.dLister = append(f.dLister, d)
+	// Bare client says it IS deleted. This should be presumed more up-to-date.
```

Comment:
I think both interpretations of "deleted" have merit. My interpretation is that some client already issued the DELETE request and it succeeded. Almost everything should consider a controller with non-nil DeletionTimestamp as not existing anymore, so calling it "deleted" in the past tense reinforces that.